packages/core/src/TimeField/TimeFieldRoot.vue

```diff
@@ -0,0 +1,320 @@
+<script lang="ts">
+import { type DateValue, Time, getLocalTimeZone, isEqualDay, toCalendarDateTime, today } from '@internationalized/date'
+
+import type { Ref } from 'vue'
+import type { PrimitiveProps } from '@/Primitive'
+import { type Formatter, createContext, useDateFormatter, useDirection, useKbd, useLocale } from '@/shared'
+import {
+  type HourCycle,
+  type SegmentPart,
+  type SegmentValueObj,
+  type TimeValue,
+  createContent,
+  getDefaultTime,
+  getTimeFieldSegmentElements,
+  initializeTimeSegmentValues,
+  isSegmentNavigationKey,
+  syncTimeSegmentValues,
+
+} from '@/shared/date'
+import { isBefore } from '@/date'
+import type { Direction, FormFieldProps } from '@/shared/types'
+
+type TimeFieldRootContext = {
+  locale: Ref<string>
+  modelValue: Ref<DateValue | undefined>
+  placeholder: Ref<DateValue>
+  isInvalid: Ref<boolean>
+  disabled: Ref<boolean>
+  readonly: Ref<boolean>
+  formatter: Formatter
+  hourCycle: HourCycle
```

Comment:
I think the `hourCycle` is abit different from https://react-spectrum.adobe.com/react-spectrum/TimeField.html#hour-cycle  when it comes to example `00:24`. 

Maybe we should change from `24:24` to `00:24`